crates/sui-indexer/src/fetcher/event_handler.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use sui_json_rpc_types::EventPage;
+use sui_sdk::SuiClient;
+use sui_types::event::EventID;
+use sui_types::query::EventQuery;
+use tracing::{info, warn};
+
+const EVENT_PAGE_SIZE: usize = 100;
+
+pub struct EventHandler {
+    rpc_client: SuiClient,
+}
+
+impl EventHandler {
+    pub fn new(rpc_client: SuiClient) -> Self {
+        Self { rpc_client }
+    }
+
+    pub async fn run_forever(&self) {
+        // TODO: read next cursor from DB if available
+        let mut next_cursor = None;
+        loop {
+            let event_page = self.fetch_event_page(next_cursor).await;
```

Comment:
Do we want to have a minimum delay between fetches, e.g. 0.1s, if the fetched results are less than `EVENT_PAGE_SIZE`? Otherwise this may spin too quickly, and potentially overload the get_events() endpoint.

Reply:
great point! This PR is a bit big now, I will write it down and promise to follow up in a following one, Thanks!